src/Tools/dotnet-monitor/Egress/Configuration/EgressProviderConfigureNamedOptions.cs

```diff
@@ -34,7 +34,23 @@ public void Configure(string name, TOptions options)
                 IConfigurationSection providerOptionsSection = providerTypeSection.GetSection(name);
                 if (providerOptionsSection.Exists())
                 {
-                    providerOptionsSection.Bind(options);
+                    if (options is ExtensionEgressProviderOptions extensionOptions)
```

Comment:
This is where we'll need to handle `Metadata` (and anything else that doesn't have a value for a child).